content/doc/book/using/working-with-projects.adoc

```diff
@@ -28,6 +28,12 @@ Darin Pope provides a summary of the differences between Pipeline projects and f
h of the three different freestyle jobs that we had. Hello, one, we slept for 60 seconds and then echo hello to. But now I want to show you one of the big differences between freestyle and pipeline. And that is durability. And what do I mean by durability? In this case, what durability means is in the case of a restart of a Jenkins controller. Any pipeline jobs that were already running once that controller comes back up will automatically start back up from where they left off. However, freestyle jobs, don't do that. So let's see the example. I'm going to go ahead and go back up to the top and I'm going to click on F1 and build now and P1 and build now. If we're going to take a look at the output for both P1 and F2, what we're going to see, we'll take a look at P1 first. We can see that we're in our sleep 60 and if we go over to F2, we can also see that we're in our sleep 60 here. So what I'm going to do is I am going to go ahead and do a kill of the process that's running our Jenkins controller. So I'm just going to do a kill - 9.55. Okay, I forgot to do so to do so to kill - 9.51.55. All right, so now I've killed the process. In fact, we can see that if we go back in here to Jenkins, it's just setting there. There's nothing going on. So let's go ahead and restart, or Jenkins controller. And this will take just a few moments to start back up. And as this starts back up, what we're going to see, once we log back in, is that the freestyle jobs are no longer running and our pipeline job will have picked up from where it left off and then finish up. So we can see here that P1 is now starting back up. We don't see any activity on F1 or F2, but if we take a look at the output of job number 2, which was running when we killed the process, we can see that we were in a sleep 60 and then we can see that we did a resuming build after a Jenkins restart, it finished waiting, we waited for everything to finish up, and then it went ahead, picked up and continued on in ran Hello2. If we take a look at our F2, we were into number 2, but we killed it during the time that 2 was running. We can also see in F3 that it never even got to that stage because after the restart, freestyle jobs have no way of knowing what was running before the shutdown. Down in the description of this video is a link to this page that explains why pipeline. We went through one of these in this video and that is the first one durability. In previous videos, I've also gone through possible and versatile. Look down in the description and you'll see some links off to those videos as well. If you have any questions or comments, you can reach out to us on Twitter at CloudBees. If this video was helpful to you, give us a thumbs up. And if you haven't subscribed to CloudBeesTV yet, why not? Take a moment, click on that subscribe button, and then ring that bell, and you'll be notified anytime there's new content available on CloudBeesTV. Thanks for watching, and we will see you in the next video.
```

Comment:
```suggestion
If you've been around Jenkins for any amount of time, you might remember the time before Pipeline existed.
In fact, you might even still have a few freestyle jobs running on your controller.
Have you ever taken the time to really understand the difference between freestyle and Pipeline?
In this video, we're going to demonstrate one of the reasons why you might want to go ahead and migrate those freestyle jobs over to Pipeline.
Here's today's starting point; I have a Jenkins LTS controller version 2.319.2 and attached to this controller, I have a Linux based agent.
Now, what we're going to be doing is we're going to be creating two jobs: a set of freestyle jobs that is going to look like and act like a single Pipeline job.
So let's get started on our freestyle jobs.

I'm going to select *New Item*, and I'm going to call this F1, select Freestyle project, and click OK.
Now, these are going to be very contrived and simple examples, but I've seen these over time to where people have made really complex freestyle jobs and chained them together, to end up with what looks like an X-like a Pipeline, but with some very, very different types of management.
So, let's take a look at how we're building this one out.
We're going to add a `build` step, select execute shell, because we have a Linux based agent, and the command is going to be `echo 'hello1'`.
I'm going to click on save.
Let's go back and create another job called F2, select freestyle, and click OK.
For this one, we're also going to add a `build` step and I'm going to say `sleep 60` and click save.
Finally, I'm going to create another job, call it F3 and select freestyle.
Here my step is now going to be `echo 'hello2'`.

So we've created three different freestyle jobs.
The first one is saying `echo 'hello1'`, the second one is `sleep 60`, then the third one is going to be `echo 'hello2'`.
Now I want to go back and verify my F1 here real quick and make sure that everything is set up correctly.
What we're doing is `echo 'hello1'`.
Now what I want to do is I want to link these three together.
The way that I need to do that is, I need to go over to F2 and I'm going to click on *Configure* and I'm going to say, under *Build Triggers*, *Build after other projects are built* so I want this one to follow F1 and I'm just going to do this as a basic, I'm not going to get into all the different trigger types, just a very simple basic F1.
I'm going to click on save.
Let's go down to F3 and I want F3 to follow F2.
Are you seeing some of the pain points here?
I'm having to go through and go into the UI and make all of these changes if I'm using freestyle.
But again, until Pipeline existed, we didn't have a choice.

So let's go back and take a look at F1.
What we're going to see a downstream project for F1 is F2, which is what we would expect.
We're going to move from F1 into F2.
If we click into f2, which we can do here, project F2, we can see the upstream is F1 and the downstream is F3.
If we click on F3, we can see that the upstream is F2.
So, we have it all wired up F1 to F2 to F3.
Now what do we do?

Well let's go ahead and just run this set of jobs.
It's not a single job, but a set of jobs, to see what happens.
When we click on build now for F1, when it starts up, it's an immediate `echo 'hello1'`, no big deal, then it's triggering a new build of F2.
If I click into F2, I can see that F2 is running and it's in a `sleep 60`, we'll fast forward through this.
Now it's triggered off to do F3 and when we take a look at F3, once it goes ahead and starts, what we'll see here is that F3 completed and it said `echo 'hello2'`.
So F1 triggered F2 that triggered F3.

Now sure, I could have created a single freestyle job with three steps, but I wanted to show you how you would have to wire jobs together in order to do these things prior to Pipeline existing.
So now, let's go over and create a Pipeline job that does the same thing, but now you're going to say, okay, you're not comparing apples to apples.
This is really apples to oranges.
Yes, I agree.
I get it, I understand, but I'm trying to show you when Pipeline first came out, what the purpose of it was.
It was to help combine the separate freestyle jobs into a definition of a job that could be managed as code.
So let's go ahead and go to Pipeline and let's paste in my Pipeline.

We can see here that I have three different stages.
These stages represented each of the different freestyle jobs that we saw previously.
So I have an `echo 'hello1'`, a `sleep 60`, and an `echo 'hello2'`.
Lets go and click on save and then click on build now.
What we'll do, is we'll take a look at the log as it starts up.
Then we'll fast forward through the rest of this.
As this finished up, you can see here that the output from our single Pipeline job represents the output of each of the three different freestyle jobs that we had.
`Hello1`, we slept for 60 seconds and then `echo 'hello2'`.

But now I want to show you one of the big differences between freestyle and Pipeline, and that is durability.
What do I mean by durability?
In this case, what durability means is in the case of a restart of a Jenkins controller, any Pipeline jobs that were already running, once that controller comes back up will automatically start back up from where they left off.
However, freestyle jobs, don't do that.
So let's see the example.
I'm going to go ahead and go back up to the top and I'm going to click on F1 and build now, and P1 and build now.
If we're going to take a look at the output for both P1 and F2, what we're going to see, we'll take a look at P1 first.

We can see that we're in our `sleep 60` and if we go over to F2, we can also see that we're in our `sleep 60` here.
So what I'm going to do is I am going to go ahead and do a `kill` of the process that's running our Jenkins controller.
So I'm just going to do a `kill - 9 5155`.
Okay, I forgot to do `sudo`, `sudo kill - 9 5155`.
All right, so now I've killed the process.
In fact, we can see that if we go back in here to Jenkins, it's just sitting there, there's nothing going on.
So let's go ahead and restart our Jenkins controller.
This will take just a few moments to start back up.

As this starts back up, what we're going to see, once we log back in, is that the freestyle jobs are no longer running and our Pipeline job will have picked up from where it left off and then finish up.
So we can see here that P1 is now starting back up.
We don't see any activity on F1 or F2, but if we take a look at the output of job number 2, which was running when we killed the process, we can see that we were in a `sleep 60` and then we can see that we did a resuming build after a Jenkins restart, it finished waiting, we waited for everything to finish up, and then it went ahead, picked up, and continued on in ran `hello2`.
If we take a look at our F2, we were into number 2, but we killed it during the time that 2 was running.
We can also see in F3, that it never even got to that stage because after the restart, freestyle jobs have no way of knowing what was running before the shutdown.

Down in the description of this video is a link to this page that explains why Pipeline.
We went through one of these in this video and that is the first one durability.
In previous videos, I've also gone through possible and versatile.
Look down in the description and you'll see some links off to those videos as well.
If you have any questions or comments, you can reach out to us on Twitter @CloudBees.
If this video was helpful to you, give us a thumbs up.
If you haven't subscribed to CloudBeesTV yet, why not?
Take a moment, click on that subscribe button, and then ring that bell, and you'll be notified anytime there's new content available on CloudBeesTV.
Thanks for watching, and we will see you in the next video.
```
Revised transcript with punctuation, paragraph breaks, capitalization, and formatting.